agentendpoint/config_task.go

```diff
@@ -283,16 +285,16 @@ func (c *configTask) enforceState(ctx context.Context) {
 				continue
 			}
 			pResult := aResult.GetOsPolicyResults().GetResults()[i]
-			for i, r := range p.GetResources() {
-				ctx = clog.WithLabels(ctx, map[string]string{"resource_id": r.GetId()})
-				res := plcy.resources[r.GetId()]
+			for i, taskResource := range p.GetResources() {
+				ctx = clog.WithLabels(ctx, map[string]string{"resource_id": taskResource.GetId()})
+				res := plcy.resources[taskResource.GetId()]
 				// Only enforce resources that need it.
-				if !res.enforcementNeeded {
+				if !res.inDesiredState {
```

Comment:
Inverse the condition?
Same in all other places where enforcementNeeded was used before.